btc_execution/sources/reader.move

```diff
@@ -0,0 +1,57 @@
+module btc_execution::reader;
+
+use btc_execution::opcode::isOpValid;
+
+#[error]
+const EBadOpcode: vector<u8> = b"Bad opcode";
+
+
+// TODO: Follow error in btc implemetation
+#[error]
+const EBadReadData: vector<u8> = b"Invalid read script";
+
+
+public struct ScriptReader has copy, drop {
+    script: vector<u8>,
+    current_index: u64
+}
+
+public fun new(script: vector<u8>): ScriptReader {
+    ScriptReader {
+        script: script,
+        current_index: 0,
+    }
+}
+
+
+public fun read(r: &mut ScriptReader, cap: u64): vector<u8> {
+    assert!(r.readable(cap), EBadReadData);
+
+    let mut i = r.current_index;
+    let mut j = 0;
+    let mut buf = vector[];
+    while (j < cap) {
+        buf.push_back(r.script[i]);
+        j = j + 1;
+        i = i + 1;
+    };
+
+    r.current_index = j;
+    buf
+}
```

Comment:
probably we don't need this function

Reply:
I think we need a read function, because in the OP_PUSH opcode family, a new need to read more than one byte. 
For example, `OP_PUSHBYTES_20`, we need to extract 20 bytes from the script.

Reply:
OK